packages/agent/src/oidc-v2.ts

```diff
@@ -0,0 +1,689 @@
+   * mobile apps) by requiring an additional verification step during token exchange.
+   */
+  code_challenge: string;
+
+  /** The method used for the PKCE challenge (typically `S256`). Must be present if `code_challenge` is included. */
+  code_challenge_method: 'S256';
+
+  /**
+   * An ID token previously issued to the client, passed as a hint about the end-user’s current or past authenticated
+   * session with the client. Can streamline user experience if already logged in.
+   */
+  id_token_hint?: string;
+
+  /** A hint to the authorization server about the login identifier the user might use. Useful for pre-filling login information. */
+  login_hint?: string;
+
+  /** Requested Authentication Context Class Reference values. Specifies the authentication context requirements. */
+  acr_values?: string;
+
+  /** When using a PAR for secure cross device flows we use a "form_post" rather than a "direct_post" */
+  response_mode: 'direct_post';
+
+  /** Used by PFI to request VCs as input to IDV process. If present, `response_type: "vp_token""` MUST also be present */
+  presentation_definition?: any;
+
+  /** A JSON object containing the Verifier metadata values (Tim's note: from TBD KCC Repo) */
+  client_metadata?: {
+    /** Array of strings, each a DID method supported for the subject of ID Token	*/
+    subject_syntax_types_supported: string[];
+    /** Human-readable string name of the client to be presented to the end-user during authorization */
+    client_name?: string;
+    /** URI of a web page providing information about the client */
+    client_uri?: string;
+    /** URI of an image logo for the client */
+    logo_uri?: string;
+    /** Array of strings representing ways to contact people responsible for this client, typically email addresses */
+    contacts?: string[];
+    /** URI that points to a terms of service document for the client */
+    tos_uri?: string;
+    /** URI that points to a privacy policy document */
+    policy_uri?: string;
+  };
+};
+
+/**
+ * Claims specific to Web5 Connect rather than OIDC
+ * The contents of this are inserted into a JWT inside of the {@link PushedAuthRequest}.
+ */
+export type Web5ConnectRequest = {
+  /** PermissionGrants that are to be sent to the provider */
+  permission_requests: ConnectPermissionRequests;
+};
+
+// old, ignore, do not use. here for comparison.
+// export interface AuthorizationRequestObject {
+//   claims?: { id_token: { delegation_grants: any; [key: string]: any } };
+//   client_id: string;
+//   client_metadata?: { client_uri?: string; [key: string]: any };
+//   code_verifier?: string;
+//   nonce: string;
+//   redirect_uri: string;
+//   state?: string;
+//   [key: string]: any;
+// }
+
+/** An auth response that is compatible with both Web5 Connect and (hopefully, WIP) OIDC SIOPv2 */
+export type HybridAuthResponse = SIOPv2AuthResponse & Web5ConnectAuthResponse;
```

Comment:
Consider similar changes as suggested for `HybridAuthRequest`

Reply:
done